plugins/git.go

```diff
@@ -29,15 +34,24 @@ func (p *GitPlugin) DefineCommand(channels Channels) (*cobra.Command, error) {
 		Args:  cobra.MatchAll(cobra.ExactArgs(1), validGitRepoArgs),
 		Run: func(cmd *cobra.Command, args []string) {
 			log.Info().Msg("Git plugin started")
-			scanGit(args[0], channels.Items, channels.Errors)
+			scanGit(args[0], p.buildScanOptions(), channels.Items, channels.Errors)
 		},
 	}
-
+	flags := command.Flags()
+	flags.IntVar(&p.Depth, argDepth, 0, "number of commits to scan from HEAD")
 	return command, nil
 }
 
-func scanGit(path string, itemsChan chan Item, errChan chan error) {
-	fileChan, err := git.GitLog(path, "")
+func (p *GitPlugin) buildScanOptions() string {
+	options := ""
+	if p.Depth > 0 {
+		options = fmt.Sprintf("--full-history --all -n %d", p.Depth)
```

Comment:
```suggestion
		options = fmt.Sprintf("-n %d", p.Depth)
```

Reply:
only to scan current checked-out branch. can you explain `--full-history` in this context?

Reply:
> by default, _gitleaks_ `GitLog` function scans using `--full-history` and `--all` options (see: https://github.com/gitleaks/gitleaks/blob/master/detect/git/git.go#L44). The reason these options are embedded in `buildScanOptions` is to maintain this behavior

From the PR description

Reply:
additional info:
https://www.git-scm.com/docs/git-log#Documentation/git-log.txt-Defaultmode history is pruned without --full-history option

I think it makes sense to make `--all` optional for all usages of this plugin, it will be more consistent and less confusing from the user's perspective.

Reply:
ok, thanks for the info. I'm okay with `--full-history`

regarding `--all`:

we can have our own version of `--all` / `--all-branches` optional arg which will include all commits from all branches.
by default (if not provided) - without `--all` = only the currently checked-out branch

Reply:
- made `--all` optional and updated `buildScanOptions` function
- added `scanAllBranches` boolean field to `GitPlugin` struct
- limited the scope of `Depth` field in `GitPlugin` struct by changing it to `depth`